src/ResourceManager/DevSpaces/Commands.DevSpaces.Test/ScenarioTests/DevSpacesTests.ps1

```diff
@@ -0,0 +1,74 @@
+<#
+.SYNOPSIS
+Test DevSpaces stuff
+#>
+function Test-AzureRmDevSpacesController
+{
+    # Setup
+    $resourceGroupName = Get-RandomResourceGroupName
+    $kubeClusterName = Get-RandomClusterName
+    $devSpacesName = Get-DevSpacesControllerName
+	
+	$devSpaceController = New-AzureRmDevSpacesController -ResourceGroupName $resourceGroupName -Name $devSpacesName -TargetClusterName $kubeClusterName -TargetResourceGroupName $resourceGroupName
+
+	Assert-AreEqual $devSpacesName $devSpaceController.Name
+	Assert-AreEqual $resourceGroupName $devSpaceController.ResourceGroupName
+	Assert-AreEqual "Succeeded" $devSpaceController.ProvisioningState
+	Assert-AreEqual "eastus" $devSpaceController.Location
+
+	$devSpaceControllerUpdated = $devSpaceController | Update-AzureRmDevSpacesController -Tag @{ tagKey="tagValue"}
+
+	Assert-AreEqual $devSpacesName $devSpaceControllerUpdated.Name
+	Assert-AreEqual $resourceGroupName $devSpaceControllerUpdated.ResourceGroupName
+	Assert-AreEqual "Succeeded" $devSpaceControllerUpdated.ProvisioningState
+
+	$devSpaceController = Get-AzureRmDevSpacesController -ResourceGroupName $resourceGroupName -Name $devSpacesName
+
+	Assert-AreEqual $devSpacesName $devSpaceController.Name
+	Assert-AreEqual $resourceGroupName $devSpaceController.ResourceGroupName
+	Assert-AreEqual "Succeeded" $devSpaceController.ProvisioningState
+	Assert-AreEqual "eastus" $devSpaceController.Location
+
+	$deletedAzureRmDevSpace = $devSpaceController | Remove-AzureRmDevSpacesController -PassThru
+	Assert-AreEqual "True" $deletedAzureRmDevSpace
+}
+
+<#
+.SYNOPSIS
+Test DevSpaces stuff with AsJob Parameter
+#>
+function Test-TestAzureDevSpacesAsJobParameter
+{
+    # Setup
+    $resourceGroupName = Get-RandomResourceGroupName
+    $kubeClusterName = Get-RandomClusterName
+	$devSpacesName = Get-DevSpacesControllerName
+	
+	$job = New-AzureRmDevSpacesController -ResourceGroupName $resourceGroupName -Name $devSpacesName -TargetClusterName $kubeClusterName -TargetResourceGroupName $resourceGroupName -AsJob
+	$job | Wait-Job
+	$devSpaceController = $job | Receive-Job
+
+	Assert-AreEqual $devSpacesName $devSpaceController.Name
+	Assert-AreEqual $resourceGroupName $devSpaceController.ResourceGroupName
+	Assert-AreEqual "Succeeded" $devSpaceController.ProvisioningState
+	Assert-AreEqual "eastus" $devSpaceController.Location
```

Comment:
same comment as above

Reply:
DevSpaces are created at the same location as Kubernetes cluster. I have assumed in the test case that AKS cluster exists in eastus.

Reply:
same reply as above... we should create the AKS cluster if one is needed for this test.

Reply:
Rest apis are mocked and no real cluster is created.

Reply:
Done